fsw/cfe-core/src/tbl/cfe_tbl_api.c

```diff
@@ -692,261 +692,222 @@ int32 CFE_TBL_Load( CFE_TBL_Handle_t TblHandle,
     int32                       Status;
     uint32                      ThisAppId;
     CFE_TBL_LoadBuff_t         *WorkingBufferPtr;
-    CFE_TBL_RegistryRec_t      *RegRecPtr = NULL;
-    CFE_TBL_AccessDescriptor_t *AccessDescPtr;
+    CFE_TBL_AccessDescriptor_t *AccessDescPtr = &CFE_TBL_TaskData.Handles[TblHandle];
+    CFE_TBL_RegistryRec_t      *RegRecPtr = &CFE_TBL_TaskData.Registry[AccessDescPtr->RegIndex];
     char                        AppName[OS_MAX_API_NAME]={"UNKNOWN"};
-    uint16                      EventMsgType = CFE_EVS_EventType_INFORMATION;
     bool                        FirstTime = false;
 
+    /* Translate AppID of caller into App Name */
+    CFE_ES_GetAppName(AppName, ThisAppId, OS_MAX_API_NAME);
 
     /* Initialize return pointer to NULL */
     WorkingBufferPtr = NULL;
 
     /* Verify access rights and get a valid Application ID for calling App */
     Status = CFE_TBL_ValidateAccess(TblHandle, &ThisAppId);
 
-    if (Status == CFE_SUCCESS)
+    if (Status != CFE_SUCCESS)
```

Comment:
This routine mixes use of status and returns throughout that make's it hard to follow.  Could it be refactored to clarify flow?  Consider doing all the checks first to decide what work to do (or error to return), then do the work?

Reply:
There's quite a lot of conditional logic (whether loading the table from a file or memory block, whether the load is partial, etc) so can't do a lot "up-front". I've refactored to do fail=>event+return to reduce deeply-nested if struct of original code. Unsure if it could be further refactored without changing/removing behavior.